shuup/admin/modules/media/views.py

```diff
@@ -179,13 +241,14 @@ def handle_get_folder(self, data):
 
     def handle_upload(self):
         request = self.request
+        shop = get_shop(request)
         try:
             folder_id = int(request.POST.get("folder_id") or request.GET.get("folder_id") or 0)
             path = request.POST.get("path") or request.GET.get("path") or None
             if folder_id != 0:
-                folder = Folder.objects.get(pk=folder_id)
+                folder = _get_folder_query(shop).get(pk=folder_id)
             elif path:
-                folder = get_or_create_folder(path)
+                folder = get_or_create_folder(shop, path)
```

Comment:
change the super generic exception `Exception` below by `DoesNotExist` and translate string

Reply:
I think the "post" already handles this.